tests/functional/object/rgw/test_rgw_sts.py

```
@@ -524,3 +520,306 @@ def test_rgw_sts_configuration(
+
+        # Step 3: Fetch credentials using incorrect AssumeRole name - expect failure
+        logger.info("Step 3: Fetching credentials using incorrect AssumeRole name")
+        incorrect_role_arn = f"arn:aws:iam::{account_id}:role/non-existent-role"
+        logger.info(f"Assuming role with incorrect ARN: {incorrect_role_arn}")
+
+        try:
+            sts_client.assume_role(
+                RoleArn=incorrect_role_arn,
+                RoleSessionName=session_name,
+            )
+            assert False, "Expected ClientError for incorrect role ARN but got success"
```

Ruff (0.15.15)

[warning] Assertion always fails, replace with `pytest.fail()`

(PT015)